### ISSUE TITLE
iOS App Store Release Checklist

### ISSUE THREAD

**Issue by benonymity** [advplyr/audiobookshelf-app#541]
Here is a rough checklist of bugs that should be fixed and features that should be implemented on the iOS app before it comes out of beta and could go onto the App Store. Not all of them are essential, but most would be good to complete before an official release.

**Features:**
- [ ] #475 - Possibly write Swift SDK for ABS, as the entire API will need to be native
     - [ ] If this is completed, a home screen/lock screen widget would be a simple addition
- [ ] #466 - Implement on iOS to parity with Android
- [ ] #260 - Sleep Timer improvements to parity with Android
- [ ] #205 - Add setting for rewind time
- [ ] #60:
    - [ ] Mark podcast as done from player menu
    - [ ] Fewer clicks to delete books (Add delete button to downloads page?)
    - [ ] Ability to swipe on books for options

**Bugs:**
- [ ] #537 - Bring iOS to parity with Android download reporting
- [ ] #518 - Seeking improvements
- [ ] #375 - Improve bookmarks and use them offline
- [ ] #517 - Improve the home view when offline
- [ ] #477 - Fix reset on play/pause
- [ ] #398 - Finish this PR
- [ ] #372 - Finish, resolves [#239](https://github.com/advplyr/audiobookshelf-app/issues/239)
- [ ] Saved play speed resets when connecting to or switching servers
- [ ] Any other crashes not documented/reproducible
- [ ] Alphabetical scroll bar in list view

**Comment by advplyr** [comment]
For the improve bookmarks bug I don't think that is actually a bug. It wasn't reproducible. Adding offline bookmarking and syncing would be a feature but would be lower priority for release.

The continue listening shelf offline was also not a bug but I closed that one since I added that in the last release. There is one other thing I want to add before coming out of beta which is showing the offline items in the library view. Currently when you are offline you will see nothing in there and all your downloaded media will be on the home page.

Thanks for making this checklist, getting there!

**Comment by advplyr** [comment]
I just closed #375 and made #163 the main thread for offline bookmarks

**Comment by benonymity** [comment]
Great, updated those two things on the list!

**Comment by Jmanko16** [comment]
Bug, selecting chapter list and going backwards does not actually take you to that chapter, it just resets current chapter to beginning on iOS

**Comment by Jmanko16** [comment]
Consider adding download button for offline playback to app (allow audiobook file or ebook to be downloaded locally and played in another app)

**Comment by whmountains** [comment]
I would be in favor of releasing as soon as possible, and adding more features after the app store release.  The app is already incredibly useful, and at least one friend has told me that the only thing steering them away from ABS is the lack of a mobile app.

I'm guessing this option was considered and discarded.  What was the reason?  Perhaps it is easier to iterate faster with testflight.

If the app were released to the store, I would still happily stay on the testflight version to help test new features.  Probably many other users would as well.  You could release to testflight first, then to the app store a few weeks later after the bugs are worked out.

**Comment by advplyr** [comment]
I'm okay with releasing before everything on the checklist is complete, but I don't think it is ready to release right now. It's close though.

Once you release you start getting rated and reviewed. Many people are not going to see that this is an early release and under active development, so they will give a negative rating on something we are currently working on.
It's not a big deal since it will be a free app, but still preventable by not rushing out of beta. Having a poorly rated app does detract users if we want people to use the software we've been building.

**Comment by vdrover** [comment]
following, and offering encouragement to release as-is.

**Comment by benonymity** [comment]
Thanks for the bump @vdrover, just updated the list with new features and bugs. I agree that getting a release out sooner would be better, especially judging by the incessant requests for more beta space, but there are a few gamebreaking playback bugs it would probably be bad to publish with. But hopefully a release isn't too far in the future!

**Comment by ruoti** [comment]
I'm willing to help out with this in the coming weeks, once school is out for the break.

However, I think it would be good for the people who have been working on iOS to triage this list. While the bugs seem reasonable to fix before release, many of the features feel like things that could easily wait until a later time. This post is nearly a year old, and based on the number of items in the list, it would seem that the app is farther from release today than it was in February. At this point, more people might be being driven away from audiobookshelf by the lack of an app then would be driven away by a good app with some issues (what it currently is) that gets 4/5 stars in the app store.

**Comment by vdrover** [comment]
Willing to contribute a 5-star rating post-release :)

**Comment by vdrover** [comment]
Just an idea: leave the source code open source and charge a 1-time fee of $2.99 for the app as a "distribution fee". That's totally compatible with the GPL, and could easily cover the apple developer fee at a minimum.

**Comment by benonymity** [comment]
@ruoti good point, I've now separated this list into a release section and nice QoL updates. Especially since some of the bugs are only affecting a few people it might not be reasonable to wait on them for a release that is affecting a lot of people. I'll also put some time in over the holiday break to try and fix some of the bigger bugs.

**Comment by advplyr** [comment]
Thanks everyone for the comments and @benonymity for keeping a running checklist.

I would say the main blocker for iOS right now is the bugs. We are getting a lot of reports, especially in Discord, about the media player closing when the phone is asleep and you pause with bluetooth. 
I haven't been able to reproduce that but I removed some background activity yesterday from the iOS side so hopefully someone can build from source and check if that is still a bug.

Maybe iOS made some recent changes that kills background tasks.
Android has similar issues to overcome with background tasks getting killed, mostly by the battery optimization logic.

**Comment by gh0sti** [comment]
> following, and offering encouragement to release as-is.

I would not release until we know for sure that the pause/resume function works. This is a very frustrating experience on iOS users when streaming/playing from download and pausing the book for more than 5 seconds kills the app and the playback session until you reopen the app. @advplyr says he has a fix but the users who have said this is an issue really should test this to make sure it's ready to go. I believe this is a show stopper until resolved with feedback.

**Comment by advplyr** [comment]
I don't have a fix because I haven't been able to reproduce the issue, but I did make some updates.

**Comment by vdrover** [comment]
I experience the issue (not a blocker for releasing for me as progress is saved), but happy to test. I have version 0.9.66 currently.

**Comment by TeroKeso** [comment]
I just bought an iPhone 15 alongside Android. If there have been no differences between sideloading vs testflight, then I can try to reproduce the problems. playback issues are familiar from the early days of the Android app

**Comment by advplyr** [comment]
I want to add that I think we need to better handle playback errors before the release. This is commonly reported as an Abs bug when the audio file cannot be played on the device. I opened an issue for it https://github.com/advplyr/audiobookshelf-app/issues/968

**Comment by gh0sti** [comment]
> I want to add that I think we need to better handle playback errors before the release. This is commonly reported as an Abs bug when the audio file cannot be played on the device. I opened an issue for it #968

since either using ABS to convert multi mp3s to m4b or just adding m4bs from elsewhere I haven't come accross this issue on iOS. Is this due to quality of the audio file such as flac ect?

**Comment by vdrover** [comment]
Seems you're getting closer. Any estimate for release?

**Comment by jvik** [comment]
FYI: I've been using the iOS Beta for a year now, and it has been working flawless for me through ~200 hours. In my opinion it is more than stable enough for a production release and I would gladly pay for it.

**Comment by nigiriemoji** [comment]
Can you please add issue #973 to the list of bugs for iOS checklist? I feel like this is going to be a common bug for many iOS 17 and newer users

**Comment by benonymity** [comment]
Done. I think right now #973 is probably the most obvious barrier to release. Can't speak for everyone, but I encounter it on an almost daily basis. When I have some time I'll try to see what if anything changed in either the ABS code or iOS 17 that broke playback so blatantly—it should hopefully be a simple solution.

**Comment by indiefan** [comment]
Regarding #973, I’m curious if the change was related to defaultRate (and thus fixed by recent pull request). @benonymity if you’re encountering it so frequently, can you try using a build from head which includes that fix and see if it still happens? (Not sure how long it takes for changes to make it into the test flight build)

**Comment by harupong** [comment]
I tried iOS app v0.9.74-beta with iOS17 installed iPhone, and #973 seems to be fixed. Thanks a lot for a great contribution, @indiefan !!
Can't wait for the official release to the App Store 🚀

**Comment by BlwAvg** [comment]
> Has anyone an Idea when the IOS Version might release? Or if I can help without a Mac?

There is a primary developer who makes daily commits, along with several additional contributors. We are excited about the upcoming iOS support, but it requires time to implement. This is a free project, and the work is progressing at an exceptionally fast pace for a volunteer effort.

As for how you can help, you absolutely can! Visit the main page and check out the "Contributing" section for more information.
https://github.com/advplyr/audiobookshelf-app

**Comment by JuanIsOnline** [comment]
I love this service and I will wait as long as needed. My only silly thought is that the developer of the app could make some money and also humor us by just releasing something that works (as I assume the test flight app works for people now) and changing a few dollars for it. I prefer to pay for this app that has its issues than to pay for the other 2 iOS apps that even though might work, I just think ABS deserves the patronage. Just my opinion. I look on here to see what things get marked off once a week, lol.

**Comment by Syphixs** [comment]
There are now 3rd party apps selling on the appstore (like ShelfPlayer which costs 5.99 Euros). I think it would be a good Idea to release it (yes in an unfinished state), as I really want to support the original Creators and not some third party.

**Comment by zongceng** [comment]
When playing content from a podcast album in the media library that contains multiple episodes, the next episode does not automatically play after the current one finishes. It is suggested to add an autoplay feature (sequential or random playback) instead of requiring the episodes to be added to a playlist for autoplay to work.

**Comment by noahkiss** [comment]
@advplyr heads up that current TestFlight build expires in two weeks

**Comment by I-Would-Like-To-Report-A-Bug-Please** [comment]
Totally agree with @JuanIsOnline. I can pay over here with known limitations instead of using some byproducts and paying twice. I would rather support you guys directly then some other blokes :)

**Comment by UrbanCircles** [comment]
it's been a while since this hit beta. any ideas when this is going to be released to the public as a v1 on iOS?

**Comment by BlwAvg** [comment]
> it's been a while since this hit beta. any ideas when this is going to be released to the public as a v1 on iOS?

See https://github.com/advplyr/audiobookshelf-app/issues/541. 
> Here is a rough checklist of bugs that should be fixed on the iOS app before it comes out of beta and could go onto the App Store.

It looks like once the bugs listed on the checklist are complete this will be moved from beta for a full iOS release.

**Comment by DDriggs00** [comment]
It looks like only #968 and #985 remain before the ios app can release. Since the beta program is currently fill, it might be possible to release sooner by shortcutting these issues.

For #985 would removing the "sleep until end of chapter" option on ios work? The option could be re-added when the actual problem was identified and resolved.

#968 would still need to be fixed, but that's a simple error message verbosity problem, which is usually fairly quick to fix.

**Comment by BlwAvg** [comment]
@DDriggs00 below is the response from the creator in this same thread. I hope this answers your question and understand this is a volunteer effort.

> I'm okay with releasing before everything on the checklist is complete, but I don't think it is ready to release right now. It's close though.
> 
> Once you release you start getting rated and reviewed. Many people are not going to see that this is an early release and under active development, so they will give a negative rating on something we are currently working on. It's not a big deal since it will be a free app, but still preventable by not rushing out of beta. Having a poorly rated app does detract users if we want people to use the software we've been building.


I should also note that ABS is open to public PRs otherwise you can visit the contributioning section on the main ABS webpage to see how you can help out.

**Comment by indiefan** [comment]
That quote was from two years ago. 

Fwiw, as someone who uses this app daily (it's by far my most used app and it's not even close), has plenty of professional software engineering experience, and has even contributed directly to this project: there's no good reason to keep this in beta at this point. @DDriggs00 suggested work-arounds for the open issues are sound.  The critical user journeys are very polished. Obviously it's not my decision to make, and I know folks have put in a lot of hard work on this and want it to be perfect, but you should be proud to share what you've built with others. It's a really great app. Sure there will be bugs, and there's always more polish that could be done, but the users who want a self-hosted option for listening to audiobooks are going to love this app.

**Comment by DDriggs00** [comment]
Of the issues listed in the lower section, #891 and *maybe* #998 look like the only ones that would affect users on a regular basis.

**Comment by BlwAvg** [comment]
I mean the options are to wait, change the creators mind by leaving comments on this github, or submit some PRs.

**Comment by ruoti** [comment]
@BlwAvg, I think that is what they are trying to do. Influence @advplyr's decisions by posting.

I'm with @indiefan, I think it is time to release the app as is. At this point, I believe it is more detrimental to Audiobookshelf to not have a working app than it is for Audiobookshelf to have an app with some bugs. I cannot actively recommend Audiobookshelf to others because they won't be able to run the app on their phone. In contrast, the app is no more buggy than Audible's app. I have things I don't like in the app, but it isn't worse than many professional apps I also use on my phone.

I recognize that I could commit PRs. I would love to do that, but I haven't been able to find the time. Like @indiefan, I have contributed in the past, so I'm not just trying to leach off the project. I really do think it would be in the project's best interest to release an app.

**Comment by ucrbuffalo** [comment]
I agree with you about this, however, there may be reasons that Apple would not allow ABS on the App Store at this time. I don’t know what those reasons are right now, but if there are any, those should be top priority so we can get an immediate release. On Feb 2, 2025, at 4:43 PM, csudderth ***@***.***> wrote:﻿
I’ve been using the app in TestFlight for quite some time and simply haven’t had any issues to report because I’ve had a great experience. I’m sure there are remaining issues I’m not aware of, but I think the app is in an excellent state. I think this may be an example of “don’t let perfect be the enemy of good.” Plus, an initial release on the App Store could potentially attract more contributors. I think withholding the apps from a proper release could lead to the project stagnating, which would be a shame because this is an amazing piece of software.

**Comment by biolan** [comment]
I am currently saving the link on homescreen to be able to use it as a webapp but this is not the point.

can't be bothered to explain all my relatives how to do that.

looking forward to this launch

**Comment by hjk321** [comment]
This has been discussed ad nauseam in the discord and the bottom line is the core devs will not budge on this.

They seem to be under the misguided assumption that apple would for whatever reason instantly nuke their app off the face of the planet if they launched now. "Apple doesn't allow beta apps on the app store!" Fella, let me explain something to you... **It's only a beta if you call it a beta.** But the devs seem incapable of grasping this basic concept.

Like I said, this has brought up before and will be brought up again, all the way until the so-called "1.0" launches in 2038.

**Comment by TeroKeso** [comment]
I agree with the developers on the list of what should be fixed before release. The problem is that there are not many developers on the IOS side. 

If you have strong feelings about IOS become a Sponsor to advplyr. Put your money in according to your feelings. Of course, this has 100% **NO** relationship anything that he or other person will have time to fix the IOS app, but maybe he/they can buy better coffee beans. I assume they drink coffee, sorry :D

**Comment by I-Would-Like-To-Report-A-Bug-Please** [comment]
This is where balance comes in. If they release the app, they could add a "buy me a coffee" option directly. With the app released, there would be more users, for sure, so there would be more probability someone would actually do that. But it's their choice.

**Comment by BlwAvg** [comment]
https://www.theregister.com/2025/02/16/open_source_maintainers_state_of_open/

I am not sure I understand the framing of the issue. You could reach out to the trillion dollar company about how their software makes it difficult for beta apps or side loading instead of kicking the team that makes the app free to use.

**Comment by ruoti** [comment]
I think this thread is getting a little off track. There are legitimate concerns with the release schedule; this is an appropriate place to share those concerns. Discussing issues like this is not out of scope for open source projects. Many of the people that have commented have contributed to this repo. We may not be core developers, but we are acting in good faith. With that said, I agree that there may be a little bit too much angst here as well.

I would note, the repo's license allows anyone to fork this project and distribute it themselves. I'm curious what @advplyr thinks about one of those wanting to see a sooner release doing just that. It's allowed according to the license, but I'm hesitant to do so without hearing from @advplyr or at least them having a chance to respond.

**Comment by clFaster** [comment]
I wouldn’t rush the release—I don’t see any urgent need for it to go out now. If you want to use the app, you can simply build and sideload it. For others who aren’t as tech-savvy, there are free alternatives like [Plappa](https://apps.apple.com/at/app/plappa/id6475201956).

**Comment by ruoti** [comment]
Plappa isn't free for local playback. Its not really an alternative. It's clear that there is fundamental disagreement about whether the app is ready or not. I don't know if the discussion is going to change that. It's been two years, and its not clear we are closer today to release than we were 2 years ago. That is why people want to do something now. That is why I raise the possibility of having some people release an unofficial branch on the app store (which, once again, is allowed by the license).

**Comment by clFaster** [comment]
AltStore and AltServer can simplify the process by automatically refreshing the app as long as your device is sometimes (at least once per week) on the same network…

**Comment by WWoodpecker** [comment]
At least it would help the users in the EU if you would create an AltStore PAL source as suggested in the article [https://github.com/advplyr/audiobookshelf-app/issues/1346](https://github.com/advplyr/audiobookshelf-app/issues/1346), if all EU users would leave the test flight, at least there would be more space for others...

**Comment by ruoti** [comment]
@clFaster I think this ignores the fact that much of the advocation for a release isn't to allow developers to run these apps but rather for us (developers) to share ABS with our family and friends. It's one thing for me to set up a sideloaded or development app on my machine, but it is completely untenable for family and friends' devices. I would guess that most people on this thread have beta access already; the push for a real release is to support the many people that we want to bring into the ABS ecosystem, but can't due to the lack of an app to help them install.

**Comment by BlwAvg** [comment]
https://www.apple.com/feedback/

**Comment by ruoti** [comment]
@BlwAvg Apple isn't a problem here. Having a user limit for betas is reasonable. The beta program isn't meant to host largely stable software for 3+ years.

It think is fine for the developers to want to delay releasing the app generally. At the same time, there is no need to gaslight those requesting a release, acting like their requests are unreasonable and that the only problem is Apple's rules. No one is at fault here. There is just a difference in opinions on the best way to move forward. Everyone has the best interest of ABS in mind, just disagreements on how to achieve those interests.

This software is licensed under the GPL, so if someone has a real problem with the impasse, they are free to solve it by releasing their own app based on this software (even charging to cover the dev fee). That is the beauty of the GPL license. I think that acts as a nice solution for now, when we can't come to agreement.

**Comment by BlwAvg** [comment]
If you have the best interest of the ABS in mind then, to use your phrasing, stop gaslighting the developer to do something they have already made their position clear on. 

There are apps available to the masses for iOS not in beta. Try these out and share with your friends. Hell I'll even send you $25 so you can buy the app for 5 of your friends. DM (can you DM on github?) your zelle whatever.

Shelfplayer($5):
https://apps.apple.com/us/app/shelfplayer/id6475221163

plappa(free with IAP?):
https://apps.apple.com/us/app/plappa/id6475201956

**Comment by davralin** [comment]
So, I just wanted to summarize this thing, perhaps for my own sake, perhaps for someone else.
This thread is over two years old, there is twelve bugs listed as "must fix", so one bug every second month, none of the closed bugs was fixed in 2025, last closed bug was in november 2024.

Testflight apparently allows 10k users, and that has been full for at least one year.

I believe that means there won't be any new users, as the barrier for entry means people have to wait for a free slot (which I believe  means someone else gave up on waiting), or sideload the app every week.

The rationale for not releasing the app is that "it's beta"-quality, and therefore would be "review-bombed".

With that summarization, I believe the only thing we as users can do is:
* fix the bugs (which it seems like the 10 000 users currently using this app (myself included) is not capable of), 
* silently just wait for someone else to contribute, or free up testflight-slots for someone else to arrive, and then contribute.
* switch to a different app.

If we look at the recommendations that @BlwAvg gave, plappa is neither open-source, nor free.
ShelfPlayer is open-source, and costs a reasonable 5$.

Funnily enough, ShelfPlayer is a one-man-show, also listed as beta on the github, and it has a 4.4 rating, I also like this feedback:

> While the interface generally looks nice, it is actually less user friendly and less functional than the first party Audiobookshelf app (beta is unfortunately full and it’s not yet on the App Store).

The last alternative is also to fork and release the app, as @ruoti politely asked about. I see no response though, so perhaps silence means approval?

**Comment by BlwAvg** [comment]
- I can agree the iOS appear to not be a priority. The @advplyr account make commits daily and handles bugs in a relatively quick fashion. I have been helped multiple times personally with other issues. I don't care about the rational behind the lack of effort as much as I do the attempts to subvert or force the completely free project in a specific direction. My friends dont care about $5 for an app. 

- I am not sure I understand the issues with other Plappa not being free or open source? Are the audiobooks, ebooks, OS/Container running ABS, or anything else opensource?  Plappa is a $1.99 yearly subscription or $5 for a lifetime (according the app store) which seems just as reasonable as ShelfPlayer. 

- Not going to touch the fork and publishing topic with a ten foot poll.

**Comment by benonymity** [comment]
Hey guys! I'm the one who opened this issue a while ago, but I haven't touched any Audiobookshelf code in a while. 
<details>
  <summary>However, I still use the app every day.</summary>
<img src="https://github.com/user-attachments/assets/b269bb15-6440-40bc-9358-978da7dcbacf">
</details>

Personally I encounter relatively few issues with the iOS app. The one persistent issue I experience is resuming a book, which shows at time 0:00 in the player and doesn't save my progress, so I just close and reopen the player whenever this happens. But it's not super common, and isn't a huge gamebreaker. It's proved hard to debug, as it's very inconsistent and XCode alone doesn't provide a ton of feedback. Maybe someday I'll put my head down and try to fix this, but it costs me a few seconds every couple of days, so I'm not very preoccupied with fixing it as it otherwise does what I need.

Anyways, bug aside, I think that this app is overall quite usable. The limitations of apps built around webView are still annoying, but are nothing we can fix without a complete rewrite which obviously won't happen. My family uses the app quite often without issue (until they forget to update the TestFlight and have to switch to ShelfPlayer.) 

From what I recall one of the reasons to delay release is the fact that the iOS app isn't at parity with Android, but it is perfectly usable without listen history, auto sleep timer, and Carplay, though all would be nice to have. However, the cost of so many people having to go to a paid app because of the full TestFlight, as well as the pain of using TestFlight in the first place, seems to outweigh the benefit of Android parity, especially as nobody's actively work on that. It's possible I'll have some time to work on a few features over a break, but am usually quite busy with other things higher up the priority list.

I know there are some API changes coming down the pipe that might need some app changes, so perhaps that's a reason to delay a release, but I think TestFlight was designed for intensive bug-hunting and rapid iteration by a small team, where at this point I'd guess 99% of users aren't developers.

I don't see a big problem with releasing this and solving major bugs as they come up, especially as similar quality ABS apps have been released and well-reviewed, even when paid (the overlap of people who can setup ABS and unforgiving review trolls seems pretty slim). But ultimately I'm not the one with my finger on the App Store trigger, (nor am I advocating for release armed with pitchforks!), just my two cents. Thoughts, @advplyr?

**Comment by DieselTech** [comment]
I'm not sure I understand the argument about "free or opensource" when it comes to a platform like iOS. An app charging $5 is perfectly reasonable when apple themselves charges the developer to even have an account and publish an app on the appstore. The unreasonable part is expecting the dev to eat this cost just to give other people something "for free". 

> The rationale for not releasing the app is that "it's beta"-quality, and therefore would be "review-bombed".

That was an initial concern to start with when ABS was young, I'd say now that a decent chunk of time has passed it might not matter so much anymore. However, the one aspect here that does matter is that Apple does not allow "beta apps" on the normal appstore. That part isn't some made up requirement from the project. That is in their [app review guidelines](https://developer.apple.com/app-store/review/guidelines/). Specifically, there is a section calling it out, [Section 2.2](https://developer.apple.com/app-store/review/guidelines/#beta-testing) 

>Demos, betas, and trial versions of your app don’t belong on the App Store – use TestFlight instead.

Too many people are just trying to get around this by just saying "Oh just don't call it a beta, simple! Problem solved!" however that doesn't take away from the fact that doing so is against apples rules. Especially when even the beta builds in testflight have to go through an in-person review from someone at apple. Apple staff login to a demo server that is setup to test the functionality of the app. If the program crashes when they are testing it, the build can be rejected. This is covered under [section 2.1](https://developer.apple.com/app-store/review/guidelines/#app-completeness) in the guidelines where they specially say: "We will reject incomplete app bundles and binaries that crash or exhibit obvious technical problems." 

I don't think it is fair to say that because other apps are breaking the rules of apples program that we should do the same just so more people can use the app. 

Where the mobile app stands right now, I think it is reasonable to say that everything listed in the "Features" section listed at the top of this issue shouldn't hold up the app release at this point. They were added to the release checklist at the time the issue was created as "nice to haves". But now that so much time has passed, they shouldn't hold things up anymore. However, the one thing that we should wait for is the data model rewrite that is currently going on. There is a higher chance of hitting errors or even potential crashes during this transition that wouldn't occur normally.

**Comment by nerdpersonthing** [comment]
Just adding my two cents that I agree with many of the above commenters - the current iOS build is not significantly buggier than competing apps (Audible, Plappa, Shelfplayer, etc), and it would be nice to see a full release now that all of the main bugs at the top of the above list have been fixed.

Sideloading the app (while possible) is not a practical option for users who aren't tech-savvy. Anecdotally, I've tried setting this up for a few friends/family who want to use my server on Apple devices, but the requirement to re-sideload every seven days means that they have all given up on it before too long. The TestFlight beta is also very difficult to get into (I believe it filled up within minutes the last time slots opened, as it was already full when I checked roughly an hour later).

No rush - I understand these things take time - but I am in agreement that a slightly buggy release is better than none at all :)

**Comment by noahkiss** [comment]
Another app popped up that I don't see mentioned here. I can't vouch for it, just saw it on the Selfhosted newsletter. 

https://soundleafapp.com/

**Comment by CLKRUN** [comment]
> Just adding my two cents that I agree with many of the above commenters - the current iOS build is not significantly buggier than competing apps (Audible, Plappa, Shelfplayer, etc), and it would be nice to see a full release now that all of the main bugs at the top of the above list have been fixed.
> 
> Sideloading the app (while possible) is not a practical option for users who aren't tech-savvy. Anecdotally, I've tried setting this up for a few friends/family who want to use my server on Apple devices, but the requirement to re-sideload every seven days means that they have all given up on it before too long. The TestFlight beta is also very difficult to get into (I believe it filled up within minutes the last time slots opened, as it was already full when I checked roughly an hour later).
> 
> No rush - I understand these things take time - but I am in agreement that a slightly buggy release is better than none at all :)

I am so grateful that there are alternative iOS apps and that they are posted here. I recently switched from Android to iOS and think it's a shame that the official app is in TestFlight. It's really hard to get in despite I'm joined the Discord and being present.
After some testing plappa is my favourite beside the android version of Audiobookshelf.

**Comment by kgraefe** [comment]
I'm pretty happy with the [Prologue Testflight](https://www.reddit.com/r/PrologueApp/comments/1dnbtf3/prologue_v4_audiobookshelf_alpha_coming_soon/) app.

**Comment by Eragos** [comment]
> I'm pretty happy with the [Prologue Testflight](https://www.reddit.com/r/PrologueApp/comments/1dnbtf3/prologue_v4_audiobookshelf_alpha_coming_soon/) app.

But this only the half of the cake. Podcasts is missing there completely :-/

**Comment by KapUttyy** [comment]
> I'm pretty happy with the [Prologue Testflight](https://www.reddit.com/r/PrologueApp/comments/1dnbtf3/prologue_v4_audiobookshelf_alpha_coming_soon/) app.

Which is also full…

**Comment by devnoname120** [comment]
@KapUttyy Doesn't look full on my end!

**Comment by KapUttyy** [comment]
@devnoname120 you are absolutely right. Clicked the wrong link. In the first reddit post is the actual beta which is not full.

**Comment by cryptiklemur** [comment]
Perfect is the enemy of the good.

If you guys wait until the app is perfect to release, it will never get released.

**Comment by RaHoni** [comment]
With the new IOS 18.6 in the EU it is now possible to direktly install an App from the Website of the Developer. Maybe this is a way to free up Space in the Testflight until the App is out of the Beta.

Here is a Link with more Informations [https://developer.apple.com/documentation/marketplacekit/distributing-your-app-from-your-website](https://developer.apple.com/documentation/marketplacekit/distributing-your-app-from-your-website)

**Comment by samuelvisser** [comment]
Just got access to the Beta after trying for a full year. It just made me angry. This app is more stable than any other ABS client out now. Not releasing it is directly effecting how useful the entire ABS project is. Just. Release. It.

**Comment by GiladHi** [comment]
Still trying to get access to the beta version, still keep getting the message "This beta is full". When are you going to release the official version? I'm sure it's already more stable than any of the other clients...

**Comment by StunnaGunnuh** [comment]
are there still fears of "review bombing" since it's not perfect?? The only people that would DL it are ones knowing that it isnt perfect, so what is the thought process behind that??

**Comment by DieselTech** [comment]
> Still trying to get access to the beta version, still keep getting the message "This beta is full". When are you going to release the official version? I'm sure it's already more stable than any of the other clients...

1 - When it is ready
2 - It isn't. Read the thread where this subject has been beaten to death. 


Again for the people that can't read - There is a massive rewrite going on that changes the data model ABS uses. That is going to cause a ton of problems and moving out of testflight removes the devs access to debug information. There already has been a huge increase in errors just from switching the authentication backend in the server these past few versions. The number of errors since putting that update out has increased. 

Everyone asking to just release it anyways is asking the projects to go against Apples own guidelines on how apps are to be created. If you don't like the pace at which development is occurring, start submitting fixes to the repo and contributing code to fix the problems.

**Comment by cryptiklemur** [comment]
> Everyone asking to just release it anyways is asking the projects to go against Apples own guidelines on how apps are to be created

Can you elaborate further on this?

**Comment by DieselTech** [comment]
> > Everyone asking to just release it anyways is asking the projects to go against Apples own guidelines on how apps are to be created
> 
> Can you elaborate further on this?

I have multiple times already in this thread

https://github.com/advplyr/audiobookshelf-app/issues/541#issuecomment-2691795139

**Comment by cryptiklemur** [comment]
Thank you for linking directly to it, I appreciate that. There are a lot of messages in here, and that comment in particular is hidden underneath a collapsed section in the UI. Just a small counterpoint though

> However, the one aspect here that does matter is that Apple does not allow "beta apps" on the normal appstore. That part isn't some made up requirement from the project. That is in their [app review guidelines](https://developer.apple.com/app-store/review/guidelines/). Specifically, there is a section calling it out, [Section 2.2](https://developer.apple.com/app-store/review/guidelines/#beta-testing)

The app you are developing right now isn't what they mean by `beta` app. By `beta` app, they mean a persistent beta environment that you aren't going to get rid of. That isn't what you are building here. You're building an MVP. 


Also, I understand that dealing with users can be frustrating, but resorting to insults (like "Again for the people that can't read") really isn't any way to proceed for either party.

**Comment by DieselTech** [comment]
Users not reading what people have taken the time to write out and explain is an insult as well. Just because something is behind a collapsed section doesn't mean it should be ignored. People just open the thread and post without reading or thinking because they are mad they can't get a spot in testflight. They don't actually care about what is being said here, they just want in. 

The other aspect of TestFlight is it gives much more detailed error messages and helps a lot with crashes. Once the app leaves TestFlight a good bulk of that data goes away. The way Apple runs things means that troubleshooting problems can be much harder once released on the official app store. As state above, the server data model is going through a massive rewrite and it is bound to cause problems with the mobile apps that consume the API. Even just the back end auth changes have caused problems with the apps recently and that isn't anywhere near as drastic of a change as the data model will be.

**Comment by cryptiklemur** [comment]
Its not an insult, and that's a very disingenuous argument. Its not ignored, its missed, and the intent there is important. With that section collapsed, its very easy to skip over without noticing it. It also prevents users from Ctrl+F-ing anything inside of it.

We do care. The difference is that some people here disagree with the stances that have been taken, and that's okay. That's what discussions like this are for. 

You would still get those detailed error messages from people who stay on TestFlight, which i would have to imagine would be a decent number of people.

**Comment by clayreimann** [comment]
@DieselTech Is there an updated list of things that need to get fixed for a release? Discord has a `ios-beta` channel that just points to this issue and all of the items above the "quality of life" line are closed/checked off (and a bunch of the below the line items are closed too).

I've just gotten my own ABS set up and am currently on the testflight for Plappa, but would love to be able to use the 1st part app. I've got some ios and web dev experience and would be inclined to help, but don't see a clear path  to out of beta. There are currently on [3 GH issues](https://github.com/advplyr/audiobookshelf-app/issues?q=is:issue%20state:open%20label:iOS) tagged `iOS`

**Comment by noahkiss** [comment]
@DieselTech, I want to address some of your recent comments. (And for others reading, as far as I can tell, DieselTech isn't an owner or contributor, just a long-time user who's following development closely).

Your comment starting with "**Again for the people that can't read**" was pretty unfair. You have to realize, the only other mention of a "data model rewrite" in this entire thread was from you, six months ago, and it's not in the main post's checklist. Most of us aren't tracking development on Discord (you seem well informed, so I assume you are), so that information isn't common knowledge here. Sharing that context would be helpful to all of us, but the way you framed it came off as unnecessarily harsh.

A few other points worth discussing:

**On TestFlight:** You mentioned that moving out of TestFlight "removes the devs access to debug information." Just to clarify, many developers have a stable app on the App Store while continuing to run a beta program on TestFlight. Releasing the app doesn't mean the TestFlight program has to end.

**On Apple's Guidelines:** I understand your concern, but while it's ultimately the owner's decision to make, the overwhelming consensus here is that the app is stable and ready. That's why your take that it would "violate guidelines" feels like such an outlier to so many of us.

**On Contributing:** The suggestion to "start submitting fixes" is that classic 'if you don't like it, fix it yourself' gatekeeping. It's dismissive to users who lack the specific skills or time to contribute code to a large project, and it often just shuts down a valid conversation.

**On User Intent:** I think you're misreading the room when you say people "**don't actually care**" and "**just want in**." Many of the comments supporting a release are from people already using the app. For new people, this is a massive thread to parse, especially on a phone. It's easy to miss a single comment when you're wading through everything, so we should probably give them the benefit of the doubt.

We appreciate your insights, and I can understand your frustration. I hope we can all continue to communicate constructively here as we all eagerly await the arrival of ABS in the App Store, coming Summer of 2027 (okay, I had to add a bit of snark here)

**Comment by Logan-Quest** [comment]
Adding my voice to request release. I have several relatives on iPhone that I'd like to connect to my Audiobookshelf server and they would all currently need to separately download and pay for Plappa, which doesn't have as nice an interface as ABS. 

The goals precluding release have long been met. The app appears to be stable and fairly polished, let's get it into people's hands and spread the ABS love!

**Comment by clayreimann** [comment]
@Logan-Quest I don't think that there will be an official release any time soon, unfortunately. There is a shortage of developers and time and existing priorities seem to be devoted to the server. AFAIK only advplyr has a Mac from the core team. I briefly flirted with helping get the iOS release sorted but found that it was hard to get traction making improvements given the previously mentioned limited time. 

I would be remiss in not mentioning that @Vito0912 has an admirably through catalogue of [3rd partly](https://abstoolbox.vito0912.de/clients) clients. There are several free iOS options (and I'm actually working on another iOS client [TestFlight link](https://testflight.apple.com/join/X6V7p1sD) that will always have basic ABS features free as my contribution back to this excellent project)

**Comment by Vito0912** [comment]
Thanks for pinging and sorry to everyone who gets another notification.  

I do not want to join the discussion about whether the app should be published in an unfinished state, and I want to make clear that I am a contributor to ABS, nothing more (no maintainer). Please don’t take my words as the opinion of the ABS team at all.  

It’s great that you published an app yourself, and I will test and add it to the list in the next few days. However, I have to say that, especially on iOS, we are now almost at a release cycle of one (in some months even two) app per month, which is a lot, looot. I understand the frustration about not being able to join the testflight, but in my opinion, it doesn’t help if we have 20 different apps made by 20 different people, each thinking they can do the best one or create an app that will be an alternative for this time.  

<img width="500" height="283" alt="Image" src="https://imgs.xkcd.com/comics/standards.png" />

And btw. I do not say there should be no competition or third party apps, but it should be a few that get support from developers then that build a well supported app. One of the freemium app essentially already is stale. 

I personally do not contribute to the official apps and have built one app in the past (as there was literally no other third party Android app). I still don’t want to work on the official ones because the effort has to be repeated twice (due to having the native code for iOS and Android split, even tough there are already very good cross-platform frameworks. So I’m not saying don’t create new apps(!), but in my opinion, having 20 different ones doesn’t help. There are already a few open-source clients. Why not contribute to them?  - This is an open question and not directly to you clayreimann

Now we have another (non-open-source?) freemium app. I can’t say anything about its quality, and I won’t comment on the others I tested either, but in my opinion, this doesn’t make sense. Even if the apps are great - and some really are - if they are closed-source, everything depends on one developer again. I don’t see how this is any better to solve this problem, to be honest.  (It's fine to have one in general, I just don't see how this helps here at the moment)

In my opinion, if there are contributions, they should go to a few selected open-source apps, whether free or not. (One app on the list is open source and still uses a paid model - That's totally fine.)  

And yes, 20 apps is a bit exaggerated, but at the current pace, it won’t take that long tbh.
And the apps on the list are just apps I know of. If I didn't get pinged I wouldn't have known it

For those who want to discuss: I am open to any input, but please create one discussion then or on DC to discuss this. Many people are subscribed to it.

---

Edit: After messaging me I was notified that the app is indeed open source, which does not invalidate any points other the fact that it is not lost knowledge, which is very, very good, but, and everyone has a to have an own opinion on that, but the app seems to be heavily, if not fully, written with AI. I myself wouldn't contribute to such projects (regardless of whom, also if advpylr would start using only or very much AI). I didn't mention this point above, because no apps I tested used nearly as much, thus I wanted to also state this point. If a maintainer uses heavily AI...

**Comment by D-Rex-III** [comment]
I just wanted to say being on iOS and finally side loading the app. I love it. It's really the best way to experience ABS on an iPhone from my testing of apps. Good work. Regardless of how long it takes I love the service and I'm excited for when the app is considered ready for release. Until then, all gratitude for the work.

**Comment by michaelkleinhenz** [comment]
I am not an iOS developer and I have no idea how iOS build pipelines work, but has anyone thought about just taking the source, forking it and releasing it to the app store under his own account? After all, this is FOSS. 🤷

**Comment by gh0sti** [comment]
> I am not an iOS developer and I have no idea how iOS build pipelines work, but has anyone thought about just taking the source, forking it and releasing it to the app store under his own account? After all, this is FOSS. 🤷 

I mean that's what other 3rd party devs have done with ShelfPlayer and SideShelf ect.

**Comment by arcticfoxwp** [comment]
I don't want to minimize the work that has been done here, but as this app is unlikely to ever be released at this point, I want to ensure people are aware of the alternatives.

Prologue, ShelfPlayer, Bookplayer, and Plappa are the best I have found. Prologue seems to have the most active development.

**Comment by BlwAvg** [comment]
https://gist.github.com/richhickey/1563cddea1002958f96e7ba9519972d9

**Comment by staticvoidmaine** [comment]
[AudioBooth](https://apps.apple.com/us/app/audiobooth-audiobooks-player/id6753017503) is another awesome option. Have also given up on this ever being released 🤣😬

**Comment by brandonleon** [comment]
> [AudioBooth](https://apps.apple.com/us/app/audiobooth-audiobooks-player/id6753017503) is another awesome option. Have also given up on this ever being released 🤣😬

Can absolutely recommend this one, just switched from BookPlayer which is an amazing standalone player. but misses the integration with ABS, you can download books from ABS but then it's completely standalone, I love how Audiobooth integrates listing stat's and syncs position back to ABS.

**Comment by lifo9** [comment]
If you have Apple Developer account, you can create your own Testflight.
I automated the process here: https://github.com/lifo9/audiobookshelf-app.
Feel free to fork  / copy.
